Add CDN Migration guidance (#306)
alphagov/paas-product-pages: pages/content/migration-guidance.mdx

If you think that your service will face issues migrating data using the documented approaches you should [get in contact with the team](https://admin.london.cloud.service.gov.uk/support/contact-us) using our support form to discuss your needs further. 

## PaaS CDN Migration

> We implement Cloud Foundry CDN routes as an AWS Cloudfront distribution behind the scenes.

Reviewer: ```suggestion
We implement Cloud Foundry CDN routes as AWS Cloudfront distributions behind the scenes.
```

If you are migrating from a Cloud Foundry CDN route, you will need to remove the domain from your cdn-route service before you can add it to your new AWS Cloudfront distribution. If you don't do this, you will receive a 'CNAMEAlreadyExists' error. (AWS documents this issue)[https://aws.amazon.com/premiumsupport/knowledge-center/resolve-cnamealreadyexists-error/].

You can either remove the cdn-route service:

```
cf delete-service my-cdn-route
```

or update the service to remove the domain you are migrating (if you have multiple domains):

```
cf update-service my-cdn-route -c '{"domain": "<remaining domains>"}'
```

> Removing the domain and re-adding it to your new AWS Cloudfront Distribution will cause your website to go down. Unfortunately changes to AWS Cloudfront can take between 5 and 20 minutes to apply. AWS support can cutover your domain instantly, however you will still get a small amount downtime as you cutover the DNS yourself.

Reviewer 2: Suggest

> AWS support are able to transfer the CNAME between CloudFront distributions but it requires some coordination and prerequisite work. If this is a path you would like to investigate, please contact us on support.

Author: I don't think referring to them as CNAME's makes sense. That implies they are going to do something with DNS. They use 'domain name' and 'alternative domain names' in the Cloudfront interface.

Do we want to be contacted by users on this issue? There is nothing we can do about it. We will not be involved in any of the AWS coordination work. All we can do is advise them to talk to AWS and explain the process better.

How about:

```suggestion
Removing the domain and re-adding it to your new AWS Cloudfront Distribution will cause your website to go down. Unfortunately changes to AWS Cloudfront can take between 5 and 20 minutes to apply.

AWS support can transfer your domain instantly between AWS Cloudfront Distributions, however you will still get a small amount downtime as you cutover the DNS yourself. This requires some coordination and prerequisite work with AWS support.
```

Reviewer 2: Happy with (alternate) domain over CNAME. Tenants definitely need to contact us; we will need to be involved in the process because we have to initiate the transfer as the owner of the source CloudFront distribution.

Author: Okay so following on from our conversation: 

```suggestion
Removing the domain and re-adding it to your new AWS Cloudfront Distribution will cause your website to go down. Unfortunately changes to AWS Cloudfront can take between 5 and 20 minutes to apply.

AWS support can transfer your domain instantly between AWS Cloudfront Distributions, however you will still get a small amount downtime as you cutover the DNS yourself. This requires some coordination and prerequisite work with AWS support and will require contacting us on support to acquire your current Distribution ID.
```

Reviewer 2: Happy with that

If you think that your service will face issues migrating data using the documented approaches you should [get in contact with the team](https://admin.london.cloud.service.gov.uk/support/contact-us) using our support form to discuss your needs further. 

## PaaS CDN Migration

If you are migrating from a Cloud Foundry CDN route, be aware that this is implemented as an AWS Cloudfront distribution behind the scenes.

This can create an issue if you are migrating to a different Cloudfront distribution. You will receive a 'CNAMEAlreadyExists' error if you try to set up the new Cloudfront distribution before the DNS cutover.

> AWS documents this issue on the following (page)[https://aws.amazon.com/premiumsupport/knowledge-center/resolve-cnamealreadyexists-error/].

Reviewer 2: https://www.gov.uk/guidance/content-design/links

> AWS documents this issue on the following (page)[https://aws.amazon.com/premiumsupport/knowledge-center/resolve-cnamealreadyexists-error/].

Author: Updated.

This means you will need to remove the domain from your cdn-route service in Cloud Foundry before you can add it to your new AWS Cloudfront distribution.

> This can be acheived in Cloud Foundry by removing the cdn-route service. E.g.

Reviewer: I'm rejigging to make it clearer (I hope). We try not to use passive voice
```suggestion
We implement Cloud Foundry CDN routes as an AWS Cloudfront distribution behind the scenes.

If you are migrating from a Cloud Foundry CDN route, you will need to remove the domain from your cdn-route service before you can add it to your new AWS Cloudfront distribution. If you don't do this, you will receive a 'CNAMEAlreadyExists' error. (AWS documents this issue)[https://aws.amazon.com/premiumsupport/knowledge-center/resolve-cnamealreadyexists-error/].

You can either remove the cdn-route service:
```

Author: Updated. Github didn't let me commit your suggestion for some reason.

```
cf delete-service my-cdn-route
```

> or by updating the service and removing the domain you are trying to cutover (if you have multiple domains):

Reviewer: ```suggestion
or update the service to remove the domain you are migrating (if you have multiple domains):
```

```
cf update-service my-cdn-route -c '{"domain": "<remaining domains>"}'
```

> This can create downtime for your application.
> 
> In order to avoid downtime you would need to cutover your domain to something other than Cloudfront for a short time or use the wildcard approach mentioned on the link above.

Reviewer: Clarifying what "this" means
```suggestion
Migrating a domain to AWS Cloudfront can create downtime for your application.

To avoid downtime you can migrate your domain to something other than AWS Cloudfront for a short time or use the (wildcard approach provided by AWS)[https://aws.amazon.com/premiumsupport/knowledge-center/resolve-cnamealreadyexists-error/].
```
